ACMILabs/lens-reader
# #80 Add support for the Sparkfun DE2120 barcode reader

Resolves #80

We'd like to add support for scanning barcodes or QR codes as a lens tap to support printed tickets.

### Acceptance Criteria
- [x] Add support for the [Sparkfun DE2120 barcode reader](https://github.com/sparkfun/DE2120_Py)

### Relevant design files
* None

### Testing instructions
1. Set your `READER_MODEL` to `Sparkfun DE2120` in Balena environment variables
1. Set your `TARGET_API_ENDPOINT` to your local XOS: `http://192.168.x.x:8000/api/`
1. Add a barcode as a `UID` for a new Lens to XOS: http://localhost:8000/admin/exhibitions/lens/add/
1. Add your Raspberry Pi to `e__tap-reader-pi-4`
1. Plug in the USB barcode reader
1. Try scanning a barcode and note it appears in XOS: http://localhost:8000/lens/
1. Note you can turn the scanner on/off using the Lights API

```bash
# Turn off reader
curl --location 'http://192.168.1.101:8082/api/lights/' \
--header 'Content-Type: application/json' \
--data '{
    "rgb_value": [0,0,0],
    "ramp_time": 0.5,
    "cross_fade": 1.0
}'

# Turn on reader
curl --location 'http://192.168.1.101:8082/api/lights/' \
--header 'Content-Type: application/json' \
--data '{
    "rgb_value": [0,0,0],
    "ramp_time": 0.5,
    "cross_fade": 0
}'
```
